tests/granulate_utils/test_cgroups.py

```diff
@@ -1,28 +1,231 @@
-
-    with patch.object(
-        cpucgroup,
-        "read_from_control_file",
-        {
-            CpuCgroup.cfs_period_us: "100",
-            CpuCgroup.cfs_quota_us: "-1",
-        }.__getitem__,
-    ):
-        assert cpucgroup.get_cpu_limit_cores() == -1.0
+import pytest
+from pytest import TempPathFactory
+
+from granulate_utils.linux.cgroups.cgroup import CgroupCoreV1, CgroupCoreV2, get_cgroup_for_process
+from granulate_utils.linux.cgroups.cpu_controller import CpuController
+from granulate_utils.linux.cgroups.cpuacct_controller import CpuAcctController
+from granulate_utils.linux.cgroups.memory_controller import MemoryController
+
+
+# Cgroup
+def test_cgroup_sanity(tmp_path_factory: TempPathFactory):
+    tmp_dir = tmp_path_factory.mktemp("base_controller")
+    cgroup_dir = Path(tmp_dir)
+    cpu_procs = cgroup_dir / "cgroup.procs"
+
+    cpu_procs.write_text("1 2 3")
+
+    cgroup_v1 = CgroupCoreV1(cgroup_dir)
+
+    assert cgroup_v1.get_pids_in_cgroup() == set([1, 2, 3])
+    cgroup_v1.assign_process_to_cgroup(4)
+    assert cgroup_v1.get_pids_in_cgroup() == set([4])  # write_text overwrites by default.
+
+    sub_cgroup_dir = cgroup_dir / "sub_cgroup"
+    assert not sub_cgroup_dir.exists()
+    sub_cgroup = cgroup_v1.get_cgroup_in_hierarchy("dummy", "sub_cgroup")
+    assert sub_cgroup_dir.exists()
+    assert sub_cgroup is not None
+    assert sub_cgroup.has_parent_cgroup(cgroup_v1.path.name)
+
+    sub_cgroup_procs = sub_cgroup_dir / "cgroup.procs"
+    sub_cgroup.assign_process_to_cgroup(5)
+    assert sub_cgroup.get_pids_in_cgroup() == set([5])
+    assert sub_cgroup_procs.read_text() == "5"
+
+    same_cgroup = sub_cgroup.get_cgroup_in_hierarchy("dummy", sub_cgroup.path.name)
+    assert same_cgroup.path == sub_cgroup.path
+
+    parent_cgroup = sub_cgroup.get_cgroup_in_hierarchy("dummy", cgroup_v1.path.name)
+    assert parent_cgroup.path == cgroup_v1.path
+
+
+def test_cgroup_v2(tmp_path_factory: TempPathFactory):
+    tmp_dir = tmp_path_factory.mktemp("base_controller")
+    cgroup_dir = Path(tmp_dir)
+    SUB_CGROUP_NAME = "sub_cgroup"
+    sub_cgroup_dir = cgroup_dir / SUB_CGROUP_NAME
+    sub_cgroup_dir.mkdir()
+    supported_controllers = sub_cgroup_dir / "cgroup.controllers"
+    enabled_controllers = sub_cgroup_dir / "cgroup.subtree_control"
+
+    CONTROLLER_TYPE = "dummy"
+    supported_controllers.write_text(CONTROLLER_TYPE)
+    cgroup = CgroupCoreV2(cgroup_dir)
+    sub_cgroup = cgroup.get_cgroup_in_hierarchy(CONTROLLER_TYPE, SUB_CGROUP_NAME)
+    assert sub_cgroup is not None
+    assert enabled_controllers.read_text().strip() == f"+{CONTROLLER_TYPE}"
+
+    with pytest.raises(AssertionError) as exception:
+        sub_cgroup = cgroup.get_cgroup_in_hierarchy("dummy2", SUB_CGROUP_NAME)
+    assert exception.value.args[0] == "Controller not supported 'dummy2'"
+
+
+def test_get_cgroup_current_process():
+    root_path = Path("/root_path")
+    proc_fs_path = "/dummy"
```

Comment:
`proc_fs_path`? What's this name

Reply:
according to the glossary you suggested - `cgroup_path`